### PR TITLE
Add hyperparameter search resources.

### PR DESCRIPTION
**Pull request by ResidentMario**
The second of the Spell feature tutorials.

### CODE REVIEW

File: hyper/hyperparamater-search.ipynb [hunk 1]
```diff
@@ -0,0 +1,717 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The last type of hyperparameter search that Spell supports is Bayesian search. In Bayesian search a [Bayesian](https://en.wikipedia.org/wiki/Bayesian_probability) selection algorithm is used to bias the search points selected towards those that the algorithm has deemed probable based on the results of prior selections.\n",
+    "\n",
+    "This is the most complex type of hyperparameter search, which is why it's also the one with the largest number of parameters.\n",
+    "\n",
+    "`parallel-runs` controls how many runs Spell will try to schedule and execute simultaneously. If this value is set to 1, the search will be performed completely sequentially, and each run will have access to the complete prior history of runs with which to make a decision about what values to choose next. If this value were set to 18, equaling the `num-runs`, we'd schedule evey run simultaneously, essentially falling back to random search. Every other number in between offers a tradeoff between how much information the model has as its prior and how quickly the search finishes running.\n",
```
Comment (ianvonseggern1) [inline hyper/hyperparamater-search.ipynb:458]: spelling `evey`

File: hyper/hyperparamater-search.ipynb [hunk 2]
```diff
@@ -0,0 +1,717 @@
...
+    "        --conv2_filter :conv2_filter: \\\n",
+    "        --dense_layer :dense_layer: \\\n",
+    "        --dropout_3 :dropout_3:\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This launches a random hyperparameter search. In random hyperparameter search, instead of picking a value from a list, Spell will pick a value from a range of values at random. For example, `conv2_filter` is set to the range `16:128`, so Spell might randomly pick the values `17, 125, 21, 99, 101, ...`.\n",
```
Comment (ianvonseggern1) [inline hyper/hyperparamater-search.ipynb:389]: I really like the part in your blog post about why random searches are usually better than grid searches, (I've long felt like in our mission to help guide toward 'best practices' our exposure of grid search was a bit of a sore spot), do you think its worth including a sentence or two to that effect in this notebook as well?

**Comment by ianvonseggern1** [comment]
Ah one thing here, changing the path of `basic.py` will make the promo tests fail. We can either stick with the current dir, or if you like this significantly better do you mind opening up a PR in our repo as well to come to promo test to the new directory?

**Comment by ResidentMario** [comment]
Oh yeah, I didn't realize that, sorry about that. I know Trey had said that he wants to embed these materials in promo tests so I definitely want to take a closer look at our promo tests anyway.

As for the rational, I want to shift the naming of folders to be mostly based on the CLI commands they correspond with, to create a tighter loop between "feature" and "demos of feature".

**Comment by ResidentMario** [comment]
Will update to reflect not needing to log in anymore in a separate future PR, once the `rsync` PR lands.

**Pull request merged**